app/concepts/todo/create.rb

```diff
@@ -1,50 +1,38 @@
 class Todo::Create < Trailblazer::Operation
+  include Model
+  model Todo, :create
+
+  contract do
+    property :title, validates: { presence: true }
+  end
 
-  # Logic for creating a TODO
-  #     if `todo_list` is specified
-  #         create the `todo`
-  #         associated with `todo_list`
-  #     else
-  #         create the `todo`
-  #         create a `user`
-  #         create a default `todo_list`
-  #         associate `todo_list` with `user`
-  #         associate `todo` with `todo_list`
   def process(params)
-    # make sure title is not blank
-    return invalid! if params[:todo][:title].blank?
+    validate(params[:todo]) do
+      contract.model.list = get_or_create_todo_list(params)
+      contract.model.list.user = get_or_create_user(params)
+      contract.model.list.save
+      contract.save
+    end
+  end
```

Comment:
Well done! A few more tweaks and it will be perfect. Currently, lines 11 to 13 sets up the todo_list, saves it, and associates it to the `model`. Those are 3 responsibilities that can be logically grouped into one method. Can you refactor this further to remove that responsibility from `process`?